src/components/TreeNode.tsx

```diff
@@ -210,9 +214,28 @@ const TreeNode = ({
   // Exception: The cursor thought and its previous siblings may temporarily be out of the viewport, such as if when New Subthought is activated on a long context. In this case, the new thought will be created below the viewport and needs to be rendered in order for scrollCursorIntoView to be activated.
   // Render virtualized thoughts with their estimated height so that document height is relatively stable.
   // Perform this check here instead of in virtualThoughtsPositioned since it changes with the scroll position (though currently `sizes` will change as new thoughts are rendered, causing virtualThoughtsPositioned to re-render anyway).
-  if (belowCursor && !isCursor && y > viewportBottom + height) {
-    return null
-  }
+  // if (belowCursor && !isCursor && y > viewportBottom + height) {
+  //   return null
+  // }
+  useEffect(() => {
+    if (isCursor && outerDivRef.current) {
+      // Clone both divs individually and combine them into one component
+      const clonedOuterDiv = outerDivRef.current.cloneNode(true) as HTMLElement
+
+      const bulletSpan = clonedOuterDiv.querySelector('span[aria-label="bullet"]')
+      if (bulletSpan) {
+        const svgElement = bulletSpan.querySelector('svg')
+        if (svgElement) {
+          // Remove all children of the SVG (this removes the <g> element and its contents)
+          svgElement.innerHTML = ''
+        }
+      }
+
+      cursorStore.onChange({
+        treeNode: clonedOuterDiv,
+      })
```

Comment:
What's the advantage of going through the ministore back channel rather than appending it directly?

The ministore adds complexity, and we already have a lot of direct DOM couplings, but maybe there's something else I'm not considering.

Reply:
We cant append it directly (from TreeNode.tsx) because the thing we want to append is still a React element, whereas `parent.appendChild(child)` only works with DOM nodes/refs. so we use `ReactDOM.createPortal()` instead

Since portals are declarative, we place `createPortal()` inside the BulletCursorOverlay component (as it is now). The portal also needs a target to mount into, so I’m storing `targetSvg` in state inside `BulletCursorOverlay`.

For passing the cloned element around, alternatives to the ministore would be React Context or prop drilling, but functionally they’re equivalent here (just different transport mechanisms).

Reply:
Yes, I'm interested in that. It's direct DOM manipulation, but we already have so much of that anyway.

If it was a clean portal with just the mounting node, that would be one thing. But there are so many DOM nodes that are being referenced for cloning and removing, I think the declarative benefits of the portal are somewhat lost anyway.